engine/verification/assigner/engine.go

```diff
@@ -169,12 +193,34 @@ func (e *Engine) ProcessFinalizedBlock(block *flow.Block) {
 		log.Error().Err(err).Msg("could not index receipts for block")
 	}
 
+	// performs chunk assigment on each receipt and pushes the assigned chunks to the
+	// chunks queue.
 	for _, receipt := range block.Payload.Receipts {
-		e.handleExecutionReceipt(receipt, blockID)
+		chunkList, err := e.receiptChunkAssignmentWithTracing(ctx, receipt, blockID)
+		resultID := receipt.ExecutionResult.ID()
+		if err != nil {
+			log.Fatal().
+				Err(err).
+				Hex("receipt_id", logging.ID(receipt.ID())).
+				Hex("result_id", logging.ID(resultID)).
+				Hex("executor_id", logging.ID(receipt.ExecutorID)).
+				Msg("could not determine assigned chunks of the receipt")
+		}
+
+		for _, chunk := range chunkList {
+			err := e.processChunkWithTracing(ctx, chunk, resultID)
+			if err != nil {
+				log.Fatal().
+					Err(err).
+					Hex("result_id", logging.ID(resultID)).
+					Hex("chunk_id", logging.ID(chunk.ID())).
+					Uint64("chunk_index", chunk.Index).
+					Msg("could not process chunk")
+			}
+		}
 	}
 
-	// tells block consumer that it is done with this block
-	e.blockConsumerNotifier.Notify(blockID)
+	e.metrics.OnAssignerProcessFinalizedBlock(block.Header.Height)
 	log.Info().Msg("finished processing finalized block")
```

Comment:
could we also log how many chunks are stored for this block?

Reply:
for debugging purpose, we could add these two fields: `my_chunks_count` and `my_chunks_stored_count`.

`my_chunks_count` is how many chunks are assigned to me in this block. 
`my_chunks_stored_count` is how many new chunks are stored. If we are processing the same finalized block again after a restart, then `my_chunks_count` could be `10`, but `my_chunks_stored_count` would be `0`.

If sealing is halt, we will need to debug and see whether assigner has processed the next unsealed block, and assigned any chunks

Reply:
added in https://github.com/onflow/flow-go/pull/466/commits/d58c52d2c06b946b8ed9ef00a5cfd55ca0aaa145